src/plugins/discover/public/application/components/discover.tsx

```diff
@@ -79,6 +80,11 @@ export function Discover({
   }
   const { TopNavMenu } = getServices().navigation.ui;
   const { savedSearch } = opts;
+  const [toggleOn, setToggleOn] = useState(true);
```

Comment:
You need to move the useState before line 77 to get rid of the warning you mentioned. furthermore I'd suggest to name it a bit more specific, since this component will grow soon. maybe name it `toggleChartOn`, or a bit more clear what it does: `showChart` (subjectively)